Mathlib/Analysis/Convex/StronglyConvex.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2023 Chenyi Li. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chenyi Li, Ziyu Wang
+-/
+import Mathlib.Analysis.InnerProductSpace.PiL2
+
+/-!
+# Stongly convex functions from InnerProductSpace E to ℝ
+
+In this file, we define the strongly convex functions and prove the basic properties
+of a strongly convex function.
+We prove the equivalent definitions for a strongly convex functions.
+We prove the tighter bound for a strongly bound comparing with a general convex function.
+We prove that if a strongly convex function admits a feasible global minima,
+then the minima is unique.
+
+## TODO
+
+Add derivative information for a strongly convex functions, and prove some relevant properties.
+Connect with L_smooth property and prove more theorems.
+-/
+
+variable {n : Type _} [Fintype n]  {E : Type*} [NormedAddCommGroup E] [InnerProductSpace ℝ E]
+
+/-- Function f (x) is m - strongly convex iff
+  f (x) - m * ‖x‖ ^ 2 / 2 is a convex function -/
+def StronglyConvexOn (s : Set E) (m : ℝ) (f : E → ℝ) : Prop :=
+ m > 0 ∧ ConvexOn ℝ s (fun x ↦ f x - m / 2 * ‖x‖ ^ 2)
```

Comment:
I would not enforce `0 < m` as part of the definition
```suggestion
  ConvexOn ℝ s (fun x ↦ f x - m / 2 * ‖x‖ ^ 2)
```
(at least try this and see whether it's nicer)